draft-ietf-httpbis-origin-frame.md

```diff
@@ -248,15 +267,19 @@ close the connection if it is too high.
 --- back
 
 
-# Non-Normative Processing Algorithm {#algo}
+# Non-Normative Processing Algorithms {#algo}
+
+The following algorithm illustrates how a client could handle received ORIGIN_SUPPORT settings:
+
+1. Initialise the Origin Set as per {{set}}.
 
 The following algorithm illustrates how a client could handle received ORIGIN frames:
 
-1. If the client is configured to use a proxy for the connection, ignore the frame and stop processing.
-2. If the connection is not identified with the "h2" protocol identifier or another protocol that has explicitly opted into this specification, ignore the frame and stop processing.
-3. If the frame occurs upon any stream except stream 0, ignore the frame and stop processing.
-4. If any of the flags 0x1, 0x2, 0x4 or 0x8 are set, ignore the frame and stop processing.
-5. If no previous ORIGIN frame on the connection has reached this step, initialise the Origin Set as per {{set}}.
+1. If the client has not received an ORIGIN_SUPPORT setting on the connection, ignore the frame and stop processing.
+2. If the client is configured to use a proxy for the connection, ignore the frame and stop processing.
+3. If the connection is not identified with the "h2" protocol identifier or another protocol that has explicitly opted into this specification, ignore the frame and stop processing.
+4. If the frame occurs upon any stream except stream 0, ignore the frame and stop processing.
```

Comment:
Should we have it generate an error?  Also, that suggests moving this step to the first step so that we can catch more errors.

Reply:
Hmm. Possibly. Which step is 'this'?

Reply:
Step 4 (frame on streams other than 0)